pkg/oidc/provider/token.go

```diff
@@ -403,15 +403,31 @@ func (h *tokenHandler) createTokenResponse(rancherToken *v3.Token, oidcClient *v
 }
 
 func (h *tokenHandler) updateClientSecretUsedTimeStamp(oidcClient *v3.OIDCClient, clientSecretID string) interface{} {
-	patch, err := json.Marshal([]struct {
-		Op    string `json:"op"`
-		Path  string `json:"path"`
-		Value any    `json:"value"`
-	}{{
-		Op:    "add",
-		Path:  "/metadata/annotations/cattle.io.oidc-client-secret-used-" + clientSecretID,
-		Value: fmt.Sprintf("%d", h.now().Unix()),
-	}})
+	var patch []byte
+	var err error
+	if oidcClient.Annotations != nil {
```

Comment:
Q: Should we check for this specific annotation here?

Reply:
It doesn't matter which annotations are present. `add` will replace if it exists. The problem was when annotations was nil. In that case the patch with` /metadata/annotations/cattle.io.oidc-client-secret-used-` was failing because this assumes annotations is not nil